### PR TITLE
Retry withdraw on failure

### PR DESCRIPTION
**Pull request by shal**
Here is overview of general idea:

In case of failure withdraw changes state to `failing` and re-queue  withdraw request.
After 5 attempts it goes to `failed` state.

### CODE REVIEW

File: app/models/withdraw.rb [hunk 1]
```diff
@@ -12,6 +12,7 @@ class Withdraw < ApplicationRecord
                failed
                confirming].freeze
```
Comment (mnaichuk) [inline app/models/withdraw.rb:14]: Add to STATE `skipped` and `failing` states

File: app/workers/worker/withdraw_coin.rb [hunk 2]
```diff
@@ -23,7 +23,11 @@ def process(payload)
 
         if withdraw.rid.blank?
           Rails.logger.warn { "The destination address doesn't exist. Skipping..." }
-          withdraw.fail!
+          if withdraw.may_retry?
```
Comment (mnaichuk) [inline app/workers/worker/withdraw_coin.rb]: I don't think that we need to retry if withdraw doesn't have recipient address

**Review by mnaichuk — CHANGES_REQUESTED** [review]
And update API Docs where we describing `states`

File: app/workers/worker/withdraw_coin.rb (hunk 2 above)
Comment (shal) [reply app/workers/worker/withdraw_coin.rb]: Agree

**Pull request merged**